src/components/Table/TableFormatters/DetailFormatter.vue

```diff
@@ -188,11 +127,7 @@ export default {
       }
 
       if (this.formatterArgs.onClick) {
-        return this.formatterArgs.onClick(this.callbackArgs)
-      }
-
-      if (this.formatterArgs.drawer) {
-        return this.showDrawer()
+        return this.formatterArgs.onClick({ ...this.callbackArgs, detailRoute: this.getDetailRoute(), formatterArgs: this.formatterArgs })
       }
 
       if (this.preventClick) {
```

Comment:
I recommend improving the regular expressions within the `getResource()` function to handle edge cases more efficiently, such as removing spaces around "details" or adjusting case sensitivity when processing titles.

Regarding optimization concerns, consider using template literals and ES6 features to make code cleaner with regards to string concatenation. Additionally, ensuring that all necessary imports are correctly scoped can improve readability of the modules. 

Lastly, consider adding comments where appropriate to explain any changes in the documentation style and functionality for clarity.